File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>&h) {
        int n=h.size();
        int s=0,e=n-1;
        int res=0;
        while(s<e){
            int ht=min(h[s],h[e]);
            int ba=(e-s);
            int temp=ht*ba;
            res=max(temp,res);
            if(h[s]<h[e]) s++;
            else if(h[e]<h[s]) e--;
            else{
                s++;
                e--;
            }
        }
        return res;
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        i,j=0,len(height)-1
        volume = 0
        while i<j:
            temp = min(height[i],height[j]) * (j-i)
            volume = max(volume,temp)
            if height[i]<height[j]:
                i+=1
            else:
                j-=1
        return volume